### ISSUE TITLE
getSession() hangs indefinitely every hour or so

### ISSUE THREAD

**Issue by izhan** [supabase/supabase-js#824]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

`await supabase.auth.getSession()` will hang indefinitely every hour or so, until the user reloads the page. We suspect it's related to the token needing to be refreshed.

## To Reproduce

1. Sign into an application with Supabase auth running
2. Wait an hour (or however long it takes for your token to expire)
3. Reload the page and call `await supabase.auth.getSession()`
4. `getSession` will hang indefinitely

It's unclear right now whether (4) happens all the time or only some of the time. But we've been managing to reproduce pretty frequently, both locally and in prod. It's been affecting our customers as well.

Reloading the browser after (4) will immediately resolve the issue. Subsequent reloads will be fine, until an hour of inactivity in which this problem occurs again.

We've tried multiple versions of supabase-js and gotrue already, to no effect. This has been occurring for the past few weeks.

https://github.com/supabase/supabase/issues/15930 seems related but that ticket seems to be concerned with immediately after Google OAuth. Our problem occurs well after initial login.

## Expected behavior

We expect `getSession` to never hang indefinitely

## System information

- OS: [e.g. macOS, Windows] macOS
- Browser (if applies) [e.g. chrome, safari] Chrome
- Version of supabase-js: [e.g. 6.0.2] Multiple (2.24, 2.26, 2.31)
- Version of Node.js: [e.g. 10.10.0] v18.16.0

**Comment by j4w8n** [comment]
If you lower your jwt expiry time, does that also shorten the timeframe of how often this occurs?

**Comment by hf** [comment]
@izhan What "environment" is your code running in -- what browser (did you try others?), what framework, what TypeScript targets...?

We generally expect to see issues reproduced with the latest version of gotrue-js before we can continue, as some of the versions you listed are known to have issues.

**Comment by izhan** [comment]
> If you lower your jwt expiry time, does that also shorten the timeframe of how often this occurs?

Have not tried yet, can do that

> @izhan What "environment" is your code running in -- what browser (did you try others?), what framework, what TypeScript targets...?
> 
> We generally expect to see issues reproduced with the latest version of gotrue-js before we can continue, as some of the versions you listed are known to have issues.

Repro'd on Chrome, Safari, Firefox. Framework is ReactJS. TS target is ES2020.

We've tested on `supabase-js@2.31` and `gotrue-js@2.46.1` which I believe were latest. The other versions I listed were tested after following threads in https://github.com/supabase/supabase-js/issues/816 and https://github.com/supabase/supabase/issues/15930 – all of which didn't work.

We've recently downgraded back to `supabase-js@2.2.2` - the oldest known version that worked for us - and `getSession()` no longer hangs. We're gonna try this version for a few more days and monitor our logs to see if it hangs for any of our customers.

**Comment by aviadmini** [comment]
@izhan did the version downgrade work?

I most likely am facing the same issue in one of the edge cases:
when app tries to load and if Supabase backend is unreachable (e.g. local dev setup is down) then calling `await supabase.auth.getSession()` hangs. It's pretty bad in my case because I'm using Svelte auth helpers and it blocks +layout.ts which blocks the rest of the app from loading. 
Should I open a separate issue?
